test/src/test/java/org/corfudb/integration/LogReplicationSinkWriterIT.java

```diff
@@ -0,0 +1,289 @@
+     *    2). Table003 and Table004 - open on source side only
+     * 3. Verify the table records and contents after snapshot sync
+     * 4. The last 4 tables will be used for log entry sync with similar setup
+     *    1). Table005 and Table006 - open on both sides with different metadata
+     *    2). Table007 and Table008 - open on source side only
+     * 5. Verify the table records and contents after log entry sync
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void testFilterAndApplyRegistryTableEntries() throws Exception {
+        setupActiveAndStandbyCorfu();
+        openLogReplicationStatusTable();
+
+        // Initial setup for streams to replicate for both source and sink side
+        int numTableActive = 8;
+        int numTableStandby = 8;
+        mapNameToMapActive = new HashMap<>();
+        mapNameToMapStandby = new HashMap<>();
+        setupStreamsToReplicate(numTableActive, true);
+        setupStreamsToReplicate(numTableStandby, false);
+
+        /* The first 4 tables are for snapshot sync */
+
+        // Table001 and Table002 - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
+        // Table003 and Table004 - Open only in Source side. Verify that their records are replicated to Sink.
+        prepareTablesToVerify(1, 2, 4);
+
+        // Add Data for Snapshot Sync
+        writeToActive(0, NUM_WRITES);
+        // Confirm data does exist on Active Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapActive.values()) {
+            assertThat(map.count()).isEqualTo(NUM_WRITES);
+        }
+
+        // Confirm data does not exist on Standby Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapStandby.values()) {
+            assertThat(map.count()).isEqualTo(0);
+        }
+
+        // Set the plugin config file path for Source before starting the process, such that Source side
+        // log replicator could read the preset streams to replicate from STREAMS_TEST_TABLE.
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        // Switch the plugin file path to Sink side for the same reason.
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        startStandbyLogReplicator();
+
+        // Verify snapshot sync completed and cluster went into log entry sync
+        verifyInLogEntrySyncState();
+        // Verify Sink side records for Table001 and Table002 are retained, and records for Table003
+        // and Table004 are replicated.
+        // Also verify all the table's contents for snapshot sync
+        verifyTableContentsAndRecords(1, 2, 4);
+
+
+        /* The last 4 tables are for log entry sync */
+
+        // Stop LR to modify Sink side registry table in advance
+        stopActiveLogReplicator();
```

Comment:
no need to stop and start the active log replicator

Reply:
the intention here is to modify the registry tables on both side first without LR running. For Table005 and Table006 they will be opened on Sink side with different tags, which will be checked afterwards. 
```
          if (i <= splitter) {
              Assert.assertFalse(corfuRecord.getMetadata().getTableOptions().getIsFederated());
              Assert.assertTrue(corfuRecord.getMetadata().getTableOptions().getRequiresBackupSupport());
          }
```

Reply:
revisited this part and I agree that we can achieve this (by opening tables on Sink first) without stopping LR.

but I'm still a little inclined to keep the current way, because for me it looks more clear if we explicitly have this "prepare" phase to guard to things we want to check. By opening tables on Sink first, we will be relying on the underlying txn and will be tricky to debug in the future IMO

Reply:
I did not quite follow.  What will be tricky to debug?  How are we relying on the underlying transactions?  We should not stop the replicator without a reason as it is not a realistic test then.

Reply:
thanks for adding the comment